src/main/java/uk/gov/hmcts/reform/managecase/service/noc/PrepareNoCService.java

```diff
@@ -0,0 +1,208 @@
+import static uk.gov.hmcts.reform.managecase.api.errorhandling.ValidationError.CASE_TYPE_CANNOT_BE_BLANK;
+import static uk.gov.hmcts.reform.managecase.api.errorhandling.ValidationError.JURISDICTION_CANNOT_BE_BLANK;
+import static uk.gov.hmcts.reform.managecase.api.errorhandling.ValidationError.MISSING_COR_ON_THE_CASE_DATA;
+import static uk.gov.hmcts.reform.managecase.api.errorhandling.ValidationError.NO_ORGANISATION_ID_IN_ANY_ORG_POLICY;
+import static uk.gov.hmcts.reform.managecase.api.errorhandling.ValidationError.NO_ORGANISATION_POLICY_ON_CASE_DATA;
+import static uk.gov.hmcts.reform.managecase.api.errorhandling.ValidationError.NO_SOLICITOR_ORGANISATION_RECORDED_IN_ORG_POLICY;
+import static uk.gov.hmcts.reform.managecase.api.errorhandling.ValidationError.ONGOING_NOC_REQUEST;
+import static uk.gov.hmcts.reform.managecase.api.errorhandling.ValidationError.ORG_POLICY_CASE_ROLE_NOT_IN_CASE_DEFINITION;
+
+@Service
+@SuppressWarnings({"PMD.UseLocaleWithCaseConversions"})
+public class PrepareNoCService {
+    protected static final String COR_CASE_ROLE_ID = "CaseRoleId";
+    protected static final String COR_REQUEST_TIMESTAMP = "RequestTimestamp";
+    private static final JsonNodeFactory JSON_NODE_FACTORY = new JsonNodeFactory(false);
+
+    private final PrdRepository prdRepository;
+    private final SecurityUtils securityUtils;
+    private final JacksonUtils jacksonUtils;
+    private final DefinitionStoreRepository definitionStoreRepository;
+
+    @Autowired
+    public PrepareNoCService(PrdRepository prdRepository,
+                             SecurityUtils securityUtils,
+                             JacksonUtils jacksonUtils,
+                             DefinitionStoreRepository definitionStoreRepository) {
+        this.prdRepository = prdRepository;
+        this.securityUtils = securityUtils;
+        this.jacksonUtils = jacksonUtils;
+        this.definitionStoreRepository = definitionStoreRepository;
+    }
+
+    public Map<String, JsonNode> prepareNoCRequest(CaseDetails caseDetails) {
+
+        String jurisdiction = caseDetails.getJurisdiction();
+        String caseTypeId = caseDetails.getCaseTypeId();
+
+        validate(isBlank(jurisdiction), JURISDICTION_CANNOT_BE_BLANK);
+        validate(isBlank(caseTypeId), CASE_TYPE_CANNOT_BE_BLANK);
+
+        List<OrganisationPolicy> orgPolicies = findPolicies(caseDetails);
+        validate(orgPolicies.isEmpty(), NO_ORGANISATION_POLICY_ON_CASE_DATA);
+
+        String changeOfRequestFieldName = caseDetails.findChangeOrganisationRequestFieldName()
+            .orElseThrow(() -> new ValidationException(MISSING_COR_ON_THE_CASE_DATA));
+
+        Map<String, JsonNode> data = caseDetails.getData();
+
+        // check that there isn't an ongoing NoCRequest - if so this new NoCRequest must be rejected
+        String caseRoleId = getCaseRoleId(data, changeOfRequestFieldName);
+        validate(isNotBlank(caseRoleId), ONGOING_NOC_REQUEST);
+
```

Comment:
Could you extract all  List<String> caseRoles; logic in a private method? may be